crates/bevy_ecs/src/entity/index_map.rs

```diff
@@ -197,6 +351,471 @@ where
 
 impl<V: Eq> Eq for EntityIndexMap<V> {}
 
+pub struct Slice<V, S = EntityHash>(PhantomData<S>, map::Slice<Entity, V>);
```

Comment:
You need `#[repr(transparent)]` here for some of the casts to be valid.  You have it on `index_set::Slice`, just not `index_map::Slice`.